src/Morphir/Elm/Frontend.elm

```diff
@@ -2085,3 +2096,128 @@ withAccessControl isExposed a =
+                case accessControlledTpeDef.value.value of
+                    Type.TypeAliasDefinition _ tpe ->
+                        Type.collectReferences tpe
+
+                    Type.CustomTypeDefinition _ accessControlledCtors ->
+                        if accessControlledCtors.access == Access.Public then
+                            Dict.toList accessControlledCtors.value
+                                |> List.concatMap Tuple.second
+                                |> List.map
+                                    (Tuple.second
+                                        >> Type.collectReferences
+                                    )
+                                |> List.foldl Set.union Set.empty
+
+                        else
+                            Set.empty
+
+            else
+                Set.empty
+
+        extractImplicitDependencies : Set FQName.FQName -> Set Path -> Set Path
+        extractImplicitDependencies publiclyExposedTypes implicitlyExposedModules =
+            case Set.toList publiclyExposedTypes of
+                [] ->
+                    implicitlyExposedModules
+
+                tpeFQN :: otherExposedTypes ->
+                    case tpeFQN of
+                        ( pkgName, modName, _ ) ->
+                            if pkgName /= packageName then
+                                -- ignore refs from other packages
+                                extractImplicitDependencies (Set.fromList otherExposedTypes) implicitlyExposedModules
+
+                            else if Set.member modName exposedModules then
+                                -- ignore explicitly exposed modules
+                                extractImplicitDependencies (Set.fromList otherExposedTypes) implicitlyExposedModules
+
+                            else if Set.member modName implicitlyExposedModules then
+                                -- ignore already collected implicitly exposed
+                                extractImplicitDependencies (Set.fromList otherExposedTypes) implicitlyExposedModules
+
+                            else
+                                let
+                                    withNewRefs : Set FQName.FQName
+                                    withNewRefs =
+                                        lookupTypeDef tpeFQN
+                                            |> Maybe.map collectImplicitlyExposedReferencesFromDef
+                                            |> Maybe.withDefault Set.empty
+                                            |> Set.union (Set.fromList otherExposedTypes)
+                                in
+                                extractImplicitDependencies
+                                    withNewRefs
+                                    (Set.insert modName implicitlyExposedModules)
+    in
+    extractImplicitDependencies exposedTypesRef Set.empty
+        |> Debug.log "implicit dependencies"
```

Comment:
I did forgot to remove it but it reminds me to mention that this feature inadvertently exposes modules.
This may or may not be a problem, but I think it makes sense to report back to the user, at some point, that some modules have been automatically exposed.